### DISCUSSION TITLE
numpy version issue

### DISCUSSION THREAD

**Discussion by right-chan** [N1-314/Intrinsic-Curiosity-Module---PyTorch#4] [Show and tell]
After creating a new environment with conda, the default version of numpy was 1.26.4. 
However, downgrading to numpy 1.26.0 resolved issues with using gym.wrapper modules.
`pip install numpy==1.26.0`

So, you can use this.

```python
from gym.wrappers import FrameStack, ResizeObservation, GrayScaleObservation

if not color_env:
    env = GrayScaleObservation(env, keep_dim=False)
env = ResizeObservation(env, shape=42)
env = FrameStack(env, num_stack=4)
```

**Comment by HyeAnn-Lee** [comment]
3줄 요약
1. 버전 문제 지적 감사합니다!
1. 여러분, `conda install opencv`를 하시면 numpy가 자동으로 1.26.4로 downgrade됩니다! cv2도 필요한 모듈이니 설치해두십시오
1. 그리고 저는 1.26.4에서 동작합니다.

============

지적 감사합니다!
`pip install gym-super-mario-bros`를 하면 NumPy 2.0.0 버전이 같이 설치되는데,
이 상태에서 `conda install opencv`를 하면 NumPy가 1.26.4로 downgrade되네요.
(이후에 gym-super-mario-bros를 설치해도 1.26.4에서 고정)

그런데 제 경우는
말해주신 코드가 잘 동작하기는 하는데, 1.26.4에서도 잘 됩니다.

``` bash
(ICM-pytorch) (base) hyeann@mdilrg-64:~/Intrinsic-Curiosity-Module---PyTorch$ python3
Python 3.9.19 (main, May  6 2024, 19:43:03) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> numpy.__version__
'1.26.4'
>>> quit()
(ICM-pytorch) (base) hyeann@mdilrg-64:~/Intrinsic-Curiosity-Module---PyTorch$
```
터미널에 쳐보면 1.26.4라고 나오고,
`conda list -e > requirements.txt` 로 뽑아서 보면

```
...
numpy=2.0.0=pypi_0
numpy-base=1.26.4=py39h8a23956_0
...
```
입니다.
`numpy`와 `numpy-base`의 차이는 https://stackoverflow.com/questions/50699252/anaconda-environment-installing-packages-numpy-base 여기를 참고하시면 될 것 같고요, (이해 못함)

여튼 요지는, version을 내리는 것에는 동의, 하지만 1.26.4에서도 동작한다는 것이 제 의견입니다!

**Comment by right-chan** [comment]
제 경우는 최신 pytorch를 사용하고 싶어서 다음 명령어(`conda install pytorch torchvision torchaudio pytorch-cuda=11.8 -c pytorch -c nvidia` (torch version: 2.3.1))로 설치 했더니 `conda install opencv`에서 에러가 발생하네요.
pytorch와 opencv 종속성에서 충돌이 난 것 같아요.

=> 제가 torch버전을 바꾼게 문제였습니다! 1.26.4에서 동작 가능하면 그대로 사용하면 될 것 같습니다.

**Comment by HyeAnn-Lee** [comment]
굳!! 수정해서 푸시하겠습니다
